internal/crypto/cert.go

```diff
@@ -22,14 +23,28 @@ const ClientBootstrapCommonNamePrefix = "client-enrollment-"
 const AdminCommonName = "flightctl-admin"
 const DeviceCommonNamePrefix = "device:"
 
+var allowedPrefixes = [...]string{
+	DeviceCommonNamePrefix,
+	ClientBootstrapCommonNamePrefix,
+}
+
```

Comment:
I'm aware this isn't what the PR tries to address, but could we make the use of prefixes a bit more consistent?
Maybe agree on a format of `[prefix:]name` (using `:` as separator, unlike in `client-enrollment-`)? Shorten `client-enrollment` to `enrollment`? Are we even using the `AdminCommonName` and do we need prefixes for users/service acconts?

Reply:
If we are going to address this and make it consistent, can I suggest we obsolete them. Fully. 

Not in this PR of course.

This is what OU field is usually used for. 

Then the question of separator does not stand as it is a separate ASN.1 field to start off with which is added to the subject by default.